src/app/shared/stores/addons/addons.state.ts

```diff
@@ -217,6 +219,43 @@ export class AddonsState {
     );
   }
 
+  @Action(GetAuthorizedtorageOauthToken)
+  getAuthorizedStorageOauthToken(ctx: StateContext<AddonsStateModel>, action: GetAuthorizedtorageOauthToken) {
+    const state = ctx.getState();
+    ctx.patchState({
+      authorizedStorageAddons: {
+        ...state.authorizedStorageAddons,
+        isLoading: true,
+      },
+    });
+
+    return this.addonsService.getAuthorizedStorageOauthToken(action.accountId).pipe(
+      tap((addon) => {
+        ctx.setState((state) => {
+          const existing = state.authorizedStorageAddons.data.find(
+            (existingAddon: AuthorizedStorageAccountModel) => existingAddon.id === addon.id
+          );
+          const updatedData = existing
+            ? state.authorizedStorageAddons.data.map((existingAddon: AuthorizedStorageAccountModel) =>
+                existingAddon.id === addon.id ? { ...existingAddon, ...addon } : existingAddon
+              )
+            : [...state.authorizedStorageAddons.data, addon];
+
+          return {
+            ...state,
+            authorizedStorageAddons: {
+              ...state.authorizedStorageAddons,
+              data: updatedData,
+              isLoading: false,
+              error: null,
+            },
+          };
+        });
+      }),
+      catchError((error) => this.handleError(ctx, 'authorizedStorageAddons', error))
```

Comment:
```suggestion
      catchError((error) => handleSectionError(ctx, 'authorizedStorageAddons', error))
```
Use reusable `handleSectionError`.